src/main/java/com/shapesecurity/shift/es2017/scope/ScopeAnalyzer.java

```diff
@@ -407,12 +442,13 @@ private State(
             this.atsForParent = atsForParent;
             this.potentiallyVarScopedFunctionDeclarations = potentiallyVarScopedFunctionDeclarations;
             this.hasParameterExpressions = hasParameterExpressions;
+            this.counter = counter;
```

Comment:
nit: as this is the first parameter, it should be the first line of the constructor (for all constructors which take it).